### ISSUE TITLE
LINSTOR leaves orphan resources on the nodes

### ISSUE THREAD

**Issue by kvaps** [LINBIT/linstor-server#287]
Hello. We faced with issue when some linstor nodes become offline:

```
# linstor n l
Defaulted container "linstor-controller" out of: linstor-controller, kube-rbac-proxy
+---------------------------------------------------------------------------------------+
| Node                               | NodeType   | Addresses                 | State   |
|=======================================================================================|
| b-node0                            | SATELLITE  | 172.30.250.150:3367 (SSL) | OFFLINE |
| h-node0                            | SATELLITE  | 172.30.250.2:3367 (SSL)   | OFFLINE |
| i-node0                            | SATELLITE  | 172.30.250.3:3367 (SSL)   | OFFLINE |
| linstor-controller-b85954459-vkpxr | CONTROLLER | 10.111.0.177:3367 (SSL)   | Online  |
| master0                            | SATELLITE  | 172.30.250.200:3367 (SSL) | Online  |
| node0                              | SATELLITE  | 172.30.250.202:3367 (SSL) | Online  |
| node1                              | SATELLITE  | 172.30.250.203:3367 (SSL) | Online  |
| node2                              | SATELLITE  | 172.30.250.204:3367 (SSL) | Online  |
| system0                            | SATELLITE  | 172.30.250.201:3367 (SSL) | Online  |
+---------------------------------------------------------------------------------------+
```

While satellite is reachable linstor-controller is not trying to recover connection to them:
```
root@master0:~# telnet 172.30.250.150 3367
Trying 172.30.250.150...
Connected to 172.30.250.150.
Escape character is '^]'.
```

After restarting linstor-controller nodes become online, but some deleted resources remain orphan on them:
```
pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807 role:Secondary
  disk:Diskless quorum:no
  node0 connection:Connecting
  node2 connection:Connecting
```

Kernel log:
```
Apr 22 14:49:26 h-node0 kernel: drbd pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807/0 drbd1003 node0: pdsk( UpToDate -> DUnknown ) repl( Established -> Off )
Apr 22 14:49:26 h-node0 kernel: drbd pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807 node0: ack_receiver terminated
Apr 22 14:49:26 h-node0 kernel: drbd pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807 node0: Terminating ack_recv thread
Apr 22 14:49:26 h-node0 kernel: drbd pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807 node0: Terminating sender thread
Apr 22 14:49:26 h-node0 kernel: drbd pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807 node0: Starting sender thread (from drbd_r_pvc-bb9e [1112666])
Apr 22 14:49:26 h-node0 kernel: drbd pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807 node0: Connection closed
Apr 22 14:49:26 h-node0 kernel: drbd pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807 node0: conn( TearDown -> Unconnected )
Apr 22 14:49:26 h-node0 kernel: drbd pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807 node0: Restarting receiver thread
Apr 22 14:49:26 h-node0 kernel: drbd pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807 node0: conn( Unconnected -> Connecting )
Apr 22 14:49:26 h-node0 kernel: drbd pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807 node2: Preparing remote state change 1428372717
Apr 22 14:49:26 h-node0 kernel: drbd pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807 node2: Committing remote state change 1428372717 (primary_nodes=0)
Apr 22 14:49:26 h-node0 kernel: drbd pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807 node2: conn( Connected -> TearDown ) peer( Secondary -> Unknown )
Apr 22 14:49:26 h-node0 kernel: drbd pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807/0 drbd1003: quorum( yes -> no )
Apr 22 14:49:26 h-node0 kernel: drbd pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807/0 drbd1003 node2: pdsk( UpToDate -> DUnknown ) repl( Established -> Off )
Apr 22 14:49:26 h-node0 kernel: drbd pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807 node2: ack_receiver terminated
Apr 22 14:49:26 h-node0 kernel: drbd pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807 node2: Terminating ack_recv thread
Apr 22 14:49:26 h-node0 kernel: drbd pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807 node2: Terminating sender thread
Apr 22 14:49:26 h-node0 kernel: drbd pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807 node2: Starting sender thread (from drbd_r_pvc-bb9e [1112668])
Apr 22 14:49:26 h-node0 kernel: drbd pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807 node2: Connection closed
Apr 22 14:49:26 h-node0 kernel: drbd pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807 node2: conn( TearDown -> Unconnected )
Apr 22 14:49:26 h-node0 kernel: drbd pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807 node2: Restarting receiver thread
Apr 22 14:49:26 h-node0 kernel: drbd pvc-bb9e3318-5554-475c-a8b5-b78f19ffc807 node2: conn( Unconnected -> Connecting )
```

Unfortunately logs from satellite and controller are missing

cc @duckhawk

**Comment by duckhawk** [comment]
[errorlog.txt](https://github.com/LINBIT/linstor-server/files/8571112/errorlog.txt)
Example of error log from offline linstor node

**Comment by duckhawk** [comment]
[errorlog-controller.txt](https://github.com/LINBIT/linstor-server/files/8571168/errorlog-controller.txt)
Logs from controller at that moment

**Comment by kvaps** [comment]
Same again, linstor version: 1.20.3, drbd version: 9.2.1

![Screenshot 2023-02-17 at 12 21 28](https://user-images.githubusercontent.com/7556217/219637431-b91e5e13-9621-422e-928f-dade983afaea.png)

resource does not exists in linstor anymore

**Comment by ghernadi** [comment]
We fixed a bug in `v1.23.0` that could have caused this issue. The problem was that a `volume-definition delete` did not properly check if all satellites are online. If a satellite was offline during the/all `vd d` operations of a resource-definition, the resource-definition could afterwards be deleted also (since there are no volumes left), leaving the offline satellite with the orphan resource behind.

This mechanic (delete VDs before RD) is used by the CSI driver. 

Feel free to reopen this issue if this bug happens again with Linstor `v1.23.0` or higher.
